src/main/java/uk/gov/pay/ledger/queue/EventMessageDto.java

```diff
@@ -21,19 +22,24 @@ public class EventMessageDto {
 
     private Boolean live;
 
+    @NotNull
     @JsonDeserialize(using = MicrosecondPrecisionDateTimeDeserializer.class)
     private ZonedDateTime timestamp;
 
+    @NotNull
```

Comment:
We can add text to annotations so the error message is clear. With `@NonNull` annotations, the error message returned will be generic so it will be unclear what must not be null.

```
{
    "error_identifier": "GENERIC",
    "message": [
        "must not be null"
    ]
}
```

Can add something like `@NotNull(message = "Field [resource_external_id] cannot be null")` so the error message will be clear as below and in the test, we can assert for the presence of specific text `Field [resource_external_id] cannot be null` in the response

```
{
    "error_identifier": "GENERIC",
    "message": [
        "Field [resource_external_id] cannot be null"
    ]
}
```

For multiple fields, it will look like

```
{
    "error_identifier": "GENERIC",
    "message": [
        "Field [timestamp] cannot be null",
        "Field [resource_external_id] cannot be null"
    ]
}
```

Reply:
thank you!